spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/flows/NimbusTokenDecoder.java

```diff
@@ -0,0 +1,43 @@
+package com.sap.cloud.security.xsuaa.token.flows;
+
+import java.net.URI;
+
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.jwt.NimbusJwtDecoderJwkSupport;
+
+/**
+ * A default {@link VariableKeySetUriTokenDecoder} implementation using
+ * {@link NimbusJwtDecoderJwkSupport} as the implementation class.
+ */
+public class NimbusTokenDecoder implements VariableKeySetUriTokenDecoder {
+
+	private NimbusJwtDecoderJwkSupport nimbusDecoder;
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see
+	 * com.sap.cloud.address.service.xsuaa.tokenflows.TokenDecoder#setJwksURI(java.
+	 * net.URI)
+	 */
+	@Override
+	public void setJwksURI(URI keySetUri) {
```

Comment:
For me this is a critical design: 
- XsuaaAutoConfiguration configures a static instance of NimbusTokenDecoder (default BEAN has static scope) .... but what if another parallel Thread (thread-2) overwrites the JwksURI in the meanwhile, while the first thread (thread-1) calls the decode() method?